plugins/circleci/src/api/index.ts

```diff
@@ -14,82 +14,4 @@
  * limitations under the License.
  */
 
-import {
-  CircleCIOptions,
-  getMe,
-  getBuildSummaries,
-  getFullBuild,
-  postBuildActions,
-  BuildAction,
-  BuildWithSteps,
-  BuildStepAction,
-  BuildSummary,
-  GitType,
-} from 'circleci-api';
-import { createApiRef, DiscoveryApi } from '@backstage/core';
-
-export { GitType };
-export type { BuildWithSteps, BuildStepAction, BuildSummary };
-
-export const circleCIApiRef = createApiRef<CircleCIApi>({
-  id: 'plugin.circleci.service',
-  description: 'Used by the CircleCI plugin to make requests',
-});
-
-const DEFAULT_PROXY_PATH = '/circleci/api';
-
-type Options = {
-  discoveryApi: DiscoveryApi;
-  /**
-   * Path to use for requests via the proxy, defaults to /circleci/api
-   */
-  proxyPath?: string;
-};
-
-export class CircleCIApi {
-  private readonly discoveryApi: DiscoveryApi;
-  private readonly proxyPath: string;
-
-  constructor(options: Options) {
-    this.discoveryApi = options.discoveryApi;
-    this.proxyPath = options.proxyPath ?? DEFAULT_PROXY_PATH;
-  }
-
-  async retry(buildNumber: number, options: Partial<CircleCIOptions>) {
-    return postBuildActions('', buildNumber, BuildAction.RETRY, {
-      circleHost: await this.getApiUrl(),
-      ...options.vcs,
-    });
-  }
-
-  async getBuilds(
-    { limit = 10, offset = 0 }: { limit: number; offset: number },
-    options: Partial<CircleCIOptions>,
-  ) {
-    return getBuildSummaries('', {
-      options: {
-        limit,
-        offset,
-      },
-      vcs: {},
-      circleHost: await this.getApiUrl(),
-      ...options,
-    });
-  }
-
-  async getUser(options: Partial<CircleCIOptions>) {
-    return getMe('', { circleHost: await this.getApiUrl(), ...options });
-  }
-
-  async getBuild(buildNumber: number, options: Partial<CircleCIOptions>) {
-    return getFullBuild('', buildNumber, {
-      circleHost: await this.getApiUrl(),
-      ...options.vcs,
-    });
-  }
-
-  private async getApiUrl() {
-    const proxyUrl = await this.discoveryApi.getBaseUrl('proxy');
-    return proxyUrl + this.proxyPath;
-  }
-}
+export * from './CircleCIApi';
```

Comment:
The ADR says:

> Index files that are re-exporting symbols from non-index files should always enumerate all exports, that is:

So I would expect 

```typescript
export {CircleCIApi, ...} from './CircleCIApi';
```

here

Reply:
Yah makes sense!  I think the doco threw me off a bit. As I go through the code, I've definitely found the "walking up the path" useful, and when I realized in Jenkins it wasn't that way for the API (or in CircleCI), just thought I'd give a shot to better understand it. Thanks for your patience, @Fox32!  Updates added.